scalar/decimal.go

```diff
@@ -9,14 +9,15 @@ import (
 type Decimal256 struct {
 	Valid bool
 	Value decimal256.Num
+	Type  arrow.DataType // Stores precision and scale
```

Comment:
Please just store properly typed dt:
```suggestion
	Type  *arrow.Decimal256Type
```